xbmc/video/VideoInfoScanner.cpp

```diff
@@ -1990,4 +1995,35 @@ namespace VIDEO
     }
     return 0;    // didn't find anything
   }
+
+  int CVideoInfoScanner::FindVideoUsingTag(CFileItem& pItem, const ScraperPtr &scraper, CScraperUrl &url, CGUIDialogProgress *progress)
+  {
+    MOVIELIST movielist;
+    CVideoInfoDownloader imdb(scraper);
+    // Start searching imdb using title:
+    if ((pItem.IsType(".mkv") || pItem.IsType(".mp4")))
+      {
+      std::unique_ptr<CVideoTagLoaderFFmpeg> ff;
+      ff.reset( new CVideoTagLoaderFFmpeg(pItem, scraper, false) );
+      if (ff->HasInfo())
+        {
+        pItem.GetVideoInfoTag()->Reset();
+        ff->Load(*pItem.GetVideoInfoTag(), false);
+        CVideoInfoTag& tag = *pItem.GetVideoInfoTag();
+        const std::string & videoTitle = tag.GetTitle();
```

Comment:
This should be doable with less code duplication. Makethis function only return title, then do two calls to findviideo above.

Reply:
Thanks for your review. Since this has not been discussed here, I'd like to also raise an objection against my own PR. I introduced a bias in behavior for 'Movies' (compared to 'TvShows', or 'MusicVideo'). This may need to be documented that the behavior for Movies is somewhere different from the other types.

Reply:
yes, we should handle music video the same whatever logic we end up with. for shows we do not currently derive names from files, so it's expected to be different.

Reply:
I have a question as to over-all design.  With this PR, it appears that video is moving in the direction of the music library where first we do a tag scan to populate the library, then we use the populated library data to feed the scrapers.  In the case of video, at this point the tag scan is optional and thus the status of the library isn't really known.  I don't think the requirement in the music scanning to wait for scraping needs to be implemented as at least for other than episodes there isn't any real dependence on tag data from multiple videos (contrasted to music where there is implied hierarchy of artists, album artists, and albums).  I'm not advocating for any particular requirement/design, just asking the question.  ISTM that if tags exist and user setting to use the tags, then they should be used in scraping as well as in populating the library.  I'm not sure what happens when user forces a reload via the UI?

Reply:
video tags are equivalent to nfo files. used if there, but in no way mandatory. a refresh is the same as the existence of a nfo file - you are asked whether you want to use locally stored information or not. just try it, it has been in master for some months.